### DISCUSSION TITLE
Odd issue with ECL and/or EMACS (R23)

### DISCUSSION THREAD

**Discussion by SirNibble** [prirun/p50em#25] [General]
Started a few days ago and only seems to happen the first time per session when invoked without a file name.
 
ECL is ON, and my abbrev is 'EMACS    emacs %1% -ttp xterm'

```
[079] sman>emacs

ERROR raised in TRY_TERMCAP at 2342(3)/35222
(illegal character in conversion)

```
echo is then off, and the next command issued (LD in this case) gives

```
[###] ER. >Function call contains too many left brackets. (EVAL_AF)
Error in variable or command function reference. (std$cp)

```

BUT - If the first time you invoke emacs with a file name (`emacs foo.txt`) there's no problems and plain 'emacs' works thereafter.

Take out ECL from my login.cpl and it's fine.

```
ecl -on -rhist [attach_point]>.cmd_history ~
        -rb '['#'] sman>' ~
        -eb '['#'] ER. >' ~
        -nwt -nwdir

```

Getting this on both my SysAdmin account and my regular user account.

Can't think of anything I've done that's been anywhere near the EMACS or ECL installations.

Tried on both PuTTY and ZOCTerminal in VT100 mode.

**Comment by hashbackup** [comment]
Does it work if you remove ecl from login.cpl and enable ecl after
login is finished?

**Comment by SirNibble** [comment]
In that scenarion ... just 'emacs' works as expected.

Weird.

**Comment by SirNibble** [comment]
**FOUND THE CAUSE**

Been wracking my brain to think about any changes I'd made, and the comment about ECL made me remember I'd added the `-nwdir` to ECL as I was tired of removing the final '>'  on an `A *>FOOctrl-i` 

With just `-nwt` **OR** `-nwdir` invoked at login you get the above error situation, but with either alone it's business as expected.

**Comment by drboone** [comment]
I've had this problem intermittently for a long time.  Not sure if it's always been on Rev23.  I've always just used `ECL -ON`, so I don't think its options are the trigger.  The crash always rears its head when I'm in the middle of some other project, so I've never distracted to chase it down.

Apparently the terminal responded to an inquiry escape sequence but some of the response was not consumed by the application but left in the input buffer.  Timing related (e.g. between the `ESC` and the rest of the sequence) maybe?

**Comment by SirNibble** [comment]
Well it certainly threw me for a loop for a bit, I can tell you.  

Oh well, at least it's sane now.

**Comment by drboone** [comment]
It'll be back. :)

At least it always seems to resolve after one-or-few retries.

**Comment by SirNibble** [comment]
:)

Must get round to getting the Sheffield Editor installed as that's the one I used on my first PRIME (a 750, iirc) as a Thames Polytechnic student. But that's a job for the next lockdown ;)

**Comment by drboone** [comment]
The 4.8.2 binaries are easy enough to install, though terminal setup may be more difficult.  IIRC, 4.5.2 includes the terminal def compiler, but the 4.8.2 doesn't.  I think the compiled defs may be compatible.
